tests/naive.py

```diff
@@ -182,73 +193,76 @@ def stump(T_A, m, T_B=None, exclusion_zone=None, row_wise=False):
     if exclusion_zone is None:
         exclusion_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
 
-    P = np.full((l, 3), np.inf)
-    I = np.full((l, 3), -1, dtype=np.int64)
+    P = np.full((l, k + 2), np.inf, dtype=np.float64)
+    I = np.full((l, k + 2), -1, dtype=np.int64)  # two more columns are to store
+    # ... left and right top-1 matrix profile indices
 
     if row_wise:  # row-wise traversal in distance matrix
         if ignore_trivial:  # self-join
             for i in range(l):
                 apply_exclusion_zone(distance_matrix[i], i, exclusion_zone, np.inf)
 
-        for i, D in enumerate(distance_matrix):
+        for i, D in enumerate(distance_matrix):  # D: distance profile
             # self-join / AB-join: matrix proifle and indices
-            idx = np.argmin(D)
-            P[i, 0] = D[idx]
-            if P[i, 0] == np.inf:
-                idx = -1
-            I[i, 0] = idx
+            indices = np.argsort(D)[:k]
+            P[i, :k] = D[indices]
+            indices[P[i, :k] == np.inf] = -1
+            I[i, :k] = indices
 
-            # self-join: left matrix profile
+            # self-join: left matrix profile index (top-1)
             if ignore_trivial and i > 0:
                 IL = np.argmin(D[:i])
                 if D[IL] == np.inf:
                     IL = -1
-                I[i, 1] = IL
+                I[i, k] = IL
 
-            # self-join: right matrix profile
+            # self-join: right matrix profile index (top-1)
             if ignore_trivial and i < D.shape[0]:
-                IR = i + np.argmin(D[i:])  # shift argmin by `i` to get true index
+                IR = i + np.argmin(D[i:])  # shift arg by `i` to get true index
```

Comment:
Maybe: "# add offset `i` to get the true index"

Reply:
"# offset by `i` to get the true index

Reply:
but realistically, we should probably remind the user why there is an offset in the first place. What do you think? As in, what is the root of the original offset that causes us to need to correct for it. I actually don't remember 😄

Reply:
> what is the root of the original offset that causes us to need to correct for it. I actually don't remember

Please correct me if I am wrong. `D` here is the distance profile of a subsequence that starts at index `i`. To find the right matrix profile of `i`, we **cannot** do `np.argmin(D)` since we want to find minimum on the right side of `i`. So, we should search the values in `D[i:]`. Hence, the value returned by `np.argmin(D[i:])` should be corrected by adding the offset `i`.

(Note that `D[i]`, `D[i+1]`, .., `D[i+excl_zone+1]` are set to `np.inf` in self-join. So, I decided to use `D[i:]` instead of `D[i+excl_zone+1:]` as I think it is less complicated)